src/common/local_file_system.cpp

```diff
@@ -52,32 +52,45 @@ static void AssertValidFileFlags(uint8_t flags) {
 }
 
 #ifndef _WIN32
```

Comment:
I don't really know how to write Windows code - but a similar fix should be applied for that implementation too.